src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -524,24 +524,30 @@ private enum NamespaceParts
                         case SyntaxKind.OpenBracketToken:
                             if (this.IsPossibleGlobalAttributeDeclaration())
                             {
-                                // incomplete members must be processed before we add any nodes to the body:
-                                ReduceIncompleteMembers(ref pendingIncompleteMembers, ref openBraceOrSemicolon, ref body, ref initialBadNodes);
-
-                                var attribute = this.ParseAttributeDeclaration();
-                                if (!isGlobal || seen > NamespaceParts.GlobalAttributes)
+                                // Could be an attribute, or it could be a collection literal at the top level.  e.g.
+                                // `[assembly: 1].XYZ();`. While this is definitely odd code, it is totally legal (as
```

Comment:
Let's make sure we have tests for `[return: AttrType].XYZ()`.

Reply:
test exists already: TopLevelDotAccess_AttributeAmbiguity2A.